services/github/src/utils.js

```diff
@@ -173,6 +173,85 @@ export const tailLog = (log, maxLines = 250, maxPayloadLength = 30000) => {
     .join('\n');
 };
 
+/**
+ *
+ * @param {string} logString
+ * @param {number} maxPayloadLength
+ * @param {number} tailLines
+ * @returns string or null
+ */
+export const extractTailLinesFromLog = (logString, maxPayloadLength, tailLines) => {
+
+  if (tailLines === 0) {
+    return null;
+  }
+
+  const tailLog = logString.split("\n").slice(-tailLines).join("\n");
+
+  if (logString.length <= maxPayloadLength) {
```

Comment:
should this be above the previous line maybe? If whole log fits, maybe it is redundant to run (somewhat expensive) string split, slice and join?

Reply:
Ahh, yes, correct, my bad. Thanks for pointing this out.

Reply:
Just realized one thing, even if the whole logString fits within the payload, we only want to return the amount of log with the specified tail lines right, so the tailLog has been calculated before this condition.

Reply:
Nice catch, indeed!

Reply:
Have not modified the code for this.